assembly/src/assembler/instruction/mod.rs

```diff
@@ -337,12 +338,30 @@ fn push_felt(span: &mut SpanBuilder, value: Felt) {
     }
 }
 
+/// A helper function that specifies if the Range Query is inclusive or exclusive.
+fn bound_into_included_u64<I>(bound: Bound<&I>) -> u64
+where
+    I: Clone + Into<u64>,
+{
+    match bound {
+        Bound::Excluded(i) => i.clone().into().saturating_sub(1),
+        Bound::Included(i) => i.clone().into(),
+        Bound::Unbounded => u64::MAX,
+    }
+}
+
 /// Returns an error if the specified value is smaller than or equal to min or greater than or
 /// equal to max. Otherwise, returns Ok(()).
-fn validate_param<I: Ord + Into<u64>>(value: I, min: I, max: I) -> Result<(), AssemblyError> {
-    if value < min || value > max {
-        Err(AssemblyError::param_out_of_bounds(value.into(), min.into(), max.into()))
-    } else {
-        Ok(())
-    }
+fn validate_param<I, R>(value: I, range: R) -> Result<(), AssemblyError>
+where
+    I: Ord + Clone + Into<u64>,
+    R: RangeBounds<I>,
+{
+    range.contains(&value).then_some(()).ok_or_else(|| {
+        AssemblyError::param_out_of_bounds(
+            value.into(),
+            bound_into_included_u64(range.start_bound()),
```

Comment:
optional nit: the error report might be wrong if the start bound is `Bound::Unbounded` because it will display `u64::MAX` instead of `0`

Reply:
Ohh... Somehow I missed it. Anyway, what should be the most suitable way to achieve this?
- Should I add an extra boolean parameter specifying if it is a start bound? That way it is easy to get the correct Unbounded data. But it leaves some space for ambiguity. [I went ahead with this]
- Next would be to return a start and end bounds vector after passing the RangeBounds itself. But again, it would be ambiguous.
- Or is there any other better way to do it?

Reply:
A parameter will work just fine as it is hardcoded so the compiler will just optimize it away